<launch>
  <!-- Arguments -->
  <arg name="robot_name" default="kobuki"/>
  <arg name="kobuki_id" default="0"/>
  <arg name="/$(arg robot_name)/initial_pose_x" default="0"/>
  <arg name="/$(arg robot_name)/initial_pose_y" default="0"/>
  <arg name="/$(arg robot_name)/initial_pose_a" default="0"/>
  
  <!-- Move robot base and amcl -->
  <group ns="$(arg robot_name)">
    <!-- AMCL -->
    <node pkg="amcl" type="amcl" name="amcl">
      <!-- Load default parameters for turtlebot -->
      <rosparam file="$(find kobuki_multi_robot)/param/amcl.yaml" command="load"/>
      <param name="initial_pose_x" value="/$(arg robot_name)/initial_pose_x"/>
      <param name="initial_pose_y" value="/$(arg robot_name)/initial_pose_y"/>
      <param name="initial_pose_a" value="/$(arg robot_name)/initial_pose_a"/>

      <param name="odom_frame_id"  value="$(arg robot_name)/odom"/>
      <param name="base_frame_id"  value="$(arg robot_name)/base_footprint"/>

      <!-- Use the single map server -->
      <remap from="static_map" to="/static_map"/>
      <remap from="map" to="/map"/>
    </node>
  </group>
</launch>